node/src/accountant/payment_adjuster/non_unit_tests/mod.rs

```diff
@@ -292,140 +251,240 @@ fn make_payable_account(
+            zipped
+                .flat_map(|(qualified_payable, thresholds)| {
+                    make_single_qualified_payable_opt(
+                        qualified_payable,
+                        &payment_inspector,
+                        &thresholds,
+                        false,
+                        now,
+                    )
+                })
+                .collect()
+        }
+    }
+}
+
 fn try_generating_qualified_payables_and_cw_balance(
     gn: &mut ThreadRng,
-    thresholds_to_be_used: &AppliedThresholds,
     accounts_count: usize,
     now: SystemTime,
-) -> Option<(
-    u128,
-    Vec<QualifiedPayableAccount>,
-    Vec<(Wallet, PaymentThresholds)>,
-)> {
-    let payables = make_payables_according_to_thresholds_setup(
-        gn,
-        &thresholds_to_be_used,
-        accounts_count,
-        now,
-    );
+) -> Option<(u128, Vec<QualifiedPayableAccount>, AppliedThresholds)> {
+    let (payables, applied_thresholds) =
+        make_payables_according_to_thresholds_setup(gn, accounts_count, now);
 
-    let (qualified_payables, wallet_and_thresholds_pairs) =
-        try_make_qualified_payables_by_applied_thresholds(payables, &thresholds_to_be_used, now);
+    let qualified_payables =
+        try_make_qualified_payables_by_applied_thresholds(payables, &applied_thresholds, now);
 
-    let balance_average = {
-        let sum: u128 = sum_as(&qualified_payables, |account| account.balance_minor());
-        sum / accounts_count as u128
-    };
-    let cw_service_fee_balance_minor = {
-        let multiplier = 1000;
-        let max_pieces = accounts_count * multiplier;
-        let number_of_pieces = generate_usize(gn, max_pieces - 2) as u128 + 2;
-        balance_average / multiplier as u128 * number_of_pieces
-    };
-    let required_service_fee_total: u128 =
-        sum_as(&qualified_payables, |account| account.balance_minor());
+    let cw_service_fee_balance_minor =
+        pick_appropriate_cw_service_fee_balance(gn, &qualified_payables, accounts_count);
+
+    let required_service_fee_total: u128 = sum_as(&qualified_payables, |account| {
+        account.initial_balance_minor()
+    });
     if required_service_fee_total <= cw_service_fee_balance_minor {
         None
     } else {
         Some((
             cw_service_fee_balance_minor,
             qualified_payables,
-            wallet_and_thresholds_pairs,
+            applied_thresholds,
         ))
     }
 }
 
+fn pick_appropriate_cw_service_fee_balance(
+    gn: &mut ThreadRng,
+    qualified_payables: &[QualifiedPayableAccount],
+    accounts_count: usize,
+) -> u128 {
+    // Value picked empirically
+    const COEFFICIENT: usize = 1000;
+    let balance_average = sum_as(qualified_payables, |account| {
+        account.initial_balance_minor()
+    }) / accounts_count as u128;
+    let max_pieces = accounts_count * COEFFICIENT;
+    let number_of_pieces = generate_usize(gn, max_pieces - 2) as u128 + 2;
```

Comment:
It turns out that the value `2` here is also an empirically chosen value. So, it should stand right next to the constant that you created earlier in this function. This way it'll be preserved for a longer period of time without someone mistakenly modifying the values while refactoring or any other code modifications.